### PR TITLE
Store Run Status

### PR DESCRIPTION
**Pull request by JamesVaughan**
Addresses issue #35.
Implements the storage of error messages or a completion status into a model system run's directory.

Implements the Validation check when constructing the model system.
Implements the RuntimeValidation check for the model system.

### CODE REVIEW

File: src/XTMF2/RunResults.cs [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+            }
+        }
+
+        /// <summary>
+        /// Store that the run completed successfully.
+        /// </summary>
+        /// <param name="runDirectory">The directory that the run was completed in.</param>
+        /// <returns></returns>
+        internal static void WriteRunCompelted(string runDirectory)
```
Comment (BrendanReilly) [inline src/XTMF2/RunResults.cs]: Minor, just spelling on completed

File: tests/XTMF2.UnitTests/Modules/FailsAtRuntime.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    You should have received a copy of the GNU General Public License
+    along with XTMF2.  If not, see <http://www.gnu.org/licenses/>.
+*/
+using System;
+using System.Collections.Generic;
+using System.Text;
+using XTMF2;
+
+namespace TestXTMF.Modules
```
Comment (BrendanReilly) [inline tests/XTMF2.UnitTests/Modules/FailsAtRuntime.cs]: Is this namespace better xtmf2.testing ? Or TestXTMF2

File: tests/XTMF2.UnitTests/TestRun.cs [hunk 3]
```diff
@@ -216,32 +214,225 @@ public void ReportRunProgress()
...
+                    string error = null;
+                    bool success = false;
+                    using SemaphoreSlim sim = new SemaphoreSlim(0);
+
+                    runBus.ClientFinishedModelSystem += (sender, e) =>
+                    {
+                        success = true;
+                        sim.Release();
+                    };
+                    runBus.ClientErrorWhenRunningModelSystem += (sender, runId, e, stack) =>
```
Comment (BrendanReilly) [inline tests/XTMF2.UnitTests/TestRun.cs:317]: Does it make sense to use OnErrorXYZ language to show it's an event?

**Review by BrendanReilly — APPROVED** [review]
Approved, comments not necessary changes

File: tests/XTMF2.UnitTests/TestRun.cs (hunk 3 above)
Comment (JamesVaughan) [reply tests/XTMF2.UnitTests/TestRun.cs:317]: Following Microsoft style for WPF it looks like events do not get the OnXYZ pattern. On is included for methods for those events that can be overridden.

**Review by BrendanReilly — APPROVED** [review]

**Pull request merged**